cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/batch/sinks/DBSink.java

```diff
@@ -135,24 +120,19 @@ public void transform(StructuredRecord input, Emitter<KeyValue<DBRecord, NullWri
     emitter.emit(new KeyValue<DBRecord, NullWritable>(new DBRecord(input, resultSetMetadata), null));
   }
 
-  private void setResultSetMetadata(Map<String, String> runtimeArgs) throws SQLException {
-    String connectionString = runtimeArgs.get(Properties.DB.CONNECTION_STRING);
-    String tableName = runtimeArgs.get(Properties.DB.TABLE_NAME);
-    String columns = runtimeArgs.containsKey(Properties.DB.COLUMNS) ? runtimeArgs.get(Properties.DB.COLUMNS) : "*";
-    String userName = runtimeArgs.get(Properties.DB.USER);
-    String password = runtimeArgs.get(Properties.DB.PASSWORD);
-
+  private void setResultSetMetadata() throws SQLException {
     Connection connection;
-    if (userName == null) {
-      connection = DriverManager.getConnection(connectionString);
+    if (dbSinkConfig.user == null) {
+      connection = DriverManager.getConnection(dbSinkConfig.connectionString);
```

Comment:
Hum, that's a bit strange. I guess it works because the shim was registered in the output format and somehow that gets called before this. We need to confirm the ordering and be able to explain this behavior